xc/xenops_server_xen.ml

```diff
@@ -325,7 +367,15 @@ let with_disk ~xc ~xs task disk write f = match disk with
          let device = create_vbd_frontend ~xc ~xs task frontend_domid vdi in
          finally
            (fun () ->
-              device |> block_device_of_vbd_frontend |> f
+              let block_device = device |> block_device_of_vbd_frontend in
+              let nbd_prefix = "hack|nbd:unix:" in
+              let is_nbd = String.startswith nbd_prefix block_device in
+              if is_nbd then begin
```

Comment:
I'll add a new logging message here like for xapi that prints out the block_device and the fact that we're using nbd-client.